src/mvc/model.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * The base MVC model class.
+ *
+ * @class Model
+ * @mixins Emitter
+ */
+
+CKEDITOR.define( [ 'emitter', 'utils' ], function( EmitterMixin, utils ) {
+	/**
+	 * Creates a new Model instance.
+	 *
+	 * @param {Object} [attributes] The model state attributes to be set during the instance creation.
+	 * @param {Object} [properties] The properties to be appended to the instance during creation.
+	 * @method constructor
+	 */
+	function Model( attributes, properties ) {
+		/**
+		 * The internal hash containing the model's state.
+		 *
+		 * @property _attributes
+		 * @private
+		 */
+		Object.defineProperty( this, '_attributes', {
+			value: {}
+		} );
+
+		// Extend this instance with the additional (out of state) properties.
+		if ( properties ) {
+			utils.extend( this, properties );
+		}
+
+		// Initialize the attributes.
+		if ( attributes ) {
+			this.set( attributes );
+		}
+	}
+
+	/**
+	 * @inheritdoc utils#extend
+	 */
+	Model.extend = utils.extendMixin;
+
+	utils.extend( Model.prototype, EmitterMixin, {
+		/**
+		 * Creates and sets the value of a model property of this object. This property will be part of the model state
+		 * and will be observable.
+		 *
+		 * It accepts also a single object literal containing key/value pairs with properties to be set.
+		 *
+		 * @param {String} name The property name.
```

Comment:
Nah... my mistake.

"attribute" is a basic concept of models related to the model state. They're clearly different from properties, which are said to not be part of the state being simply a js feature. Therefore I'm all for using the correct name consistently in a way to differentiate them.

It's a matter of getting used to it and then things will sound very clear for us.

I'll fix this.